scripts/common/object_store.sh

```diff
@@ -204,49 +167,53 @@ EOF
+    # Update velero to use new object store only if currently using object store since velero may have already been
     # updated to use an off-cluster object store.
     if kubernetes_resource_exists velero backupstoragelocation default; then
-        echo "Updating velero to use minio"
+        echo "Updating velero to use new object store $destination_host"
         s3Url=$(kubectl -n velero get backupstoragelocation default -ojsonpath='{ .spec.config.s3Url }')
-        if [ "$s3Url" = "http://${RGW_HOST}" ]; then
-            kubectl -n velero patch backupstoragelocation default --type=merge -p "{\"spec\":{\"config\":{\"s3Url\":\"http://${MINIO_HOST}\",\"publicUrl\":\"http://${minioIP}\"}}}"
+        if [ "$s3Url" = "http://${source_host}" ]; then
+            kubectl -n velero patch backupstoragelocation default --type=merge -p "{\"spec\":{\"config\":{\"s3Url\":\"http://${destination_host}\",\"publicUrl\":\"http://${newIP}\"}}}"
 
             while read -r resticrepo; do
                 oldResticIdentifier=$(kubectl -n velero get resticrepositories "$resticrepo" -ojsonpath="{ .spec.resticIdentifier }")
-                newResticIdentifier=$(echo "$oldResticIdentifier" | sed "s/${RGW_HOST}/${MINIO_HOST}/")
+                newResticIdentifier=$(echo "$oldResticIdentifier" | sed "s/${source_host}/${destination_host}/")
                 kubectl -n velero patch resticrepositories "$resticrepo" --type=merge -p "{\"spec\":{\"resticIdentifier\":\"${newResticIdentifier}\"}}"
             done < <(kubectl -n velero get resticrepositories --selector=velero.io/storage-location=default --no-headers | awk '{ print $1 }')
         else
-            echo "default backupstoragelocation was not rgw, skipping"
+            echo "The Velero default backupstoragelocation was not $source_host, not updating to use $destination_host"
         fi
     fi
     if kubernetes_resource_exists velero secret cloud-credentials; then
-        if kubectl -n velero get secret cloud-credentials -ojsonpath='{ .data.cloud }' | base64 -d | grep -q "$RGW_ACCESS_KEY_ID"; then
-            kubectl -n velero get secret cloud-credentials -ojsonpath='{ .data.cloud }' | base64 -d > cloud
-            sed -i "s/aws_access_key_id=.*/aws_access_key_id=${MINIO_ACCESS_KEY_ID}/" cloud
-            sed -i "s/aws_secret_access_key=.*/aws_secret_access_key=${MINIO_ACCESS_KEY_SECRET}/" cloud
-            cloud=$(cat cloud | base64 -w 0)
+        if kubectl -n velero get secret cloud-credentials -ojsonpath='{ .data.cloud }' | base64 -d | grep -q "$source_access_key"; then
+            local temp_file=
+            temp_file=$(mktemp)
+            kubectl -n velero get secret cloud-credentials -ojsonpath='{ .data.cloud }' | base64 -d > "$temp_file"
+            sed -i "s/aws_access_key_id=.*/aws_access_key_id=${destination_access_key}/" cloud
+            sed -i "s/aws_secret_access_key=.*/aws_secret_access_key=${destination_secret_key}/" cloud
```

Comment:
Seems to me that we mixed the "cloud" and the new temp_file variable here, no ? I mean, shouldn't `cloud` be `$temp_file` in the previous two lines ?

Reply:
yes
yes it should be